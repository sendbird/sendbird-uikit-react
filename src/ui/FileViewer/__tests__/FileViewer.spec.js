import React from 'react';
import { render, screen } from '@testing-library/react';

import { FileViewerComponent as FileViewer } from "../index";
import { msg0, msg1 } from '../data.mock';

describe('ui/FileViewer', () => {
  it('should display image', function() {
    const {
      sender,
      type,
      url,
      name = 'example-text',
    } = msg0;
    const { profileUrl, nickname = '' } = sender;
    render(
      <FileViewer
        profileUrl={profileUrl}
        nickname={nickname}
        type={type}
        url={url}
        name={name}
        onClose={() => {}}
        onDelete={() => {}}
        />
    );
    expect(
      screen.getByAltText(msg0.name).className
    ).toBe('sendbird-fileviewer__content__img');
    expect(
      screen.getByAltText(msg0.name).className
    ).not.toContain('sendbird-fileviewer__content__video');
    expect(
      screen.getByAltText(msg0.name).src
    ).toEqual(msg0.url);
  });

  it('should display video', function() {
    const {
      sender,
      type,
      url,
      name = '',
    } = msg1;
    const { profileUrl, nickname = '' } = sender;
    const { container } = render(
      <FileViewer
        profileUrl={profileUrl}
        nickname={nickname}
        type={type}
        url={url}
        name={name}
        onClose={() => {}}
        onDelete={() => {}}
      />
    );
    expect(
      container.getElementsByClassName('sendbird-fileviewer__content__video')[0].className
    ).not.toContain('sendbird-fileviewer__content__img');
    expect(
      container.getElementsByClassName('sendbird-fileviewer__content__video')[0].className
    ).toBe('sendbird-fileviewer__content__video');
    expect(
      container.getElementsByClassName('sendbird-fileviewer__content__video')[0].children[0].src
    ).toEqual(url);
  });

  it('should handle unsupported msg', function() {
    const unsupportedMsg = { sender: {} };
    const profileUrl = '';
    const nickname = '';
    const {
      type = '',
      url = '',
      name = '',
    } = unsupportedMsg;
    const { container } = render(
      <FileViewer
        profileUrl={profileUrl}
        nickname={nickname}
        type={type}
        url={url}
        name={name}
        onClose={() => {}}
        onDelete={() => {}}
      />
    );
    expect(
      container.getElementsByClassName('sendbird-fileviewer__content__unsupported')[0].className
    ).toBe('sendbird-fileviewer__content__unsupported');
    expect(
      container.getElementsByClassName('sendbird-fileviewer__header__right')[0].children[0].className
    ).not.toBe('sendbird-fileviewer__header__right__actions');
  });

  it('should do a snapshot test of the FileViewer DOM', function() {
    const {
      sender,
      type,
      url,
      name = '',
    } = msg0;
    const { profileUrl, nickname = '' } = sender;
    const { asFragment } = render(
      <FileViewer
        profileUrl={profileUrl}
        nickname={nickname}
        type={type}
        url={url}
        name={name}
        onClose={() => {}}
        onDelete={() => {}}
        message={msg0}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
