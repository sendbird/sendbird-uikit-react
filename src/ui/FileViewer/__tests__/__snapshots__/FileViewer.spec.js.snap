// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui/FileViewer should do a snapshot test of the FileViewer DOM 1`] = `
<DocumentFragment>
  <div
    class="sendbird-fileviewer"
    tabindex="1"
  >
    <div
      class="sendbird-fileviewer__header"
    >
      <div
        class="sendbird-fileviewer__header__left"
      >
        <div
          class="sendbird-fileviewer__header__left__avatar"
        >
          <div
            class=" sendbird-avatar"
            role="button"
            style="height: 32px; width: 32px; z-index: 0; left: 0px;"
            tabindex="0"
          >
            <div
              class="sendbird-avatar-img sendbird-image-renderer"
              style="width: 100%; min-width: min(400px, 32px); max-width: 400px;"
            >
              <div
                class="sendbird-image-renderer__image"
                style="width: 100%; min-width: min(400px, 32px); max-width: 400px; position: absolute; background-repeat: no-repeat; background-position: center; background-size: cover; background-image: url(https://static.sendbird.com/sample/profiles/profile_15_512px.png);"
              />
              <img
                alt=""
                class="sendbird-image-renderer__hidden-image-loader"
                src="https://static.sendbird.com/sample/profiles/profile_15_512px.png"
              />
            </div>
          </div>
        </div>
        <span
          class="sendbird-fileviewer__header__left__filename sendbird-label sendbird-label--h-2 sendbird-label--color-onbackground-1"
        >
          sent-mail (1).png
        </span>
        <span
          class="sendbird-fileviewer__header__left__sender-name sendbird-label sendbird-label--body-1 sendbird-label--color-onbackground-2"
        >
          hoon102
        </span>
      </div>
      <div
        class="sendbird-fileviewer__header__right"
      >
        <div
          class="sendbird-fileviewer__header__right__actions"
        >
          <a
            class="sendbird-fileviewer__header__right__actions__download"
            href="https://static.sendbird.com/sample/profiles/profile_15_512px.png"
            rel="noopener noreferrer"
            target="_blank"
          >
            <div
              class=" sendbird-icon sendbird-icon-down-load sendbird-icon-color--on-background-1"
              role="button"
              style="width: 24px; min-width: 24px; height: 24px; min-height: 24px;"
              tabindex="0"
            >
              <test-file-stub />
            </div>
          </a>
        </div>
        <div
          class="sendbird-fileviewer__header__right__actions__close"
        >
          <div
            class=" sendbird-icon sendbird-icon-close sendbird-icon-color--on-background-1"
            role="button"
            style="width: 24px; min-width: 24px; height: 24px; min-height: 24px;"
            tabindex="0"
          >
            <test-file-stub />
          </div>
        </div>
      </div>
    </div>
    <div
      class="sendbird-fileviewer__content"
    >
      <img
        alt="sent-mail (1).png"
        class="sendbird-fileviewer__content__img"
        src="https://static.sendbird.com/sample/profiles/profile_15_512px.png"
      />
    </div>
  </div>
</DocumentFragment>
`;
